v1/util/formatter_util.cc

```diff
@@ -0,0 +1,37 @@
+#include "formatter_util.h"
+
+namespace cloudevents {
+namespace formatter_util {
+
+using ::cloudevents::format::Format;
+using ::cloudevents::format::Formatter;
+using ::cloudevents::format::JsonFormatter;
+
+absl::StatusOr<Format> FormatterUtil::DestringifyFormat(
+        const std::string& format_str) {
+    if (format_str == "json") {
+        return Format::kJson;
+    }
+    return absl::InvalidArgumentError("The given format is not currently recognized by the SDK.");
+}
+
+std::string FormatterUtil::StringifyFormat(const Format& format) {
+    switch (format) {
+        case Format::kJson:
+            return std::string("json");
+    }
+}
+
+absl::StatusOr<
+        std::unique_ptr<Formatter>> 
+        FormatterUtil::GetFormatter(const Format& format) {
+    switch (format) {
+        case Format::kJson:
+            auto m = new JsonFormatter;
```

Comment:
use absl::make_unique https://abseil.io/tips/126

Reply:
The issue is that absl::StatusOr doesn't seem to be able to be able to recognize that `std::unique_pointer<JsonFormatter>` is a polymorphism of `std::unique_pointer<Formatter>`

So if I try the following code:
```
absl::StatusOr<std::unique_ptr<Formatter>> FormatterUtil::GetFormatter(
    const Format& format) {
  if (format == Format::kJson) {
    return absl::make_unique<JsonFormatter>();
  }
  return absl::InternalError(kErrUnkFormatter);
}
```
I get this error:
`v1/util/formatter_util.cc:35:44: error: could not convert 'absl::make_unique(Args&& ...) [with T = cloudevents::format::JsonFormatter; Args = {}; typename absl::memory_internal::MakeUniqueResult<T>::scalar = std::unique_ptr<cloudevents::format::JsonFormatter, std::default_delete<cloudevents::format::JsonFormatter> >]()' from 'absl::memory_internal::MakeUniqueResult<cloudevents::format::JsonFormatter>::scalar' {aka 'std::unique_ptr<cloudevents::format::JsonFormatter, std::default_delete<cloudevents::format::JsonFormatter> >'} to 'absl::StatusOr<std::unique_ptr<cloudevents::format::Formatter> >'
   35 |     return absl::make_unique<JsonFormatter>();
`

Which in short form I think reads `Could not convert std::unique_ptr<cloudevents::format::JsonFormatter to std::unique_ptr<cloudevents::format::Formatter> `

Reply:
Maybe try 
std::unique_ptr<Formatter> foo = make_unique<JsonFormatter>();
return foo;
?